editor/src/components/navigator/navigator-item/navigator-item.tsx

```diff
@@ -372,14 +373,45 @@ export const NavigatorItem: React.FunctionComponent<
     [dispatch, elementPath, isFocusableComponent],
   )
 
+  const parentPath = React.useMemo(() => EP.parentPath(props.elementPath), [props.elementPath])
+
+  const parent = useEditorState(
+    Substores.metadata,
+    (store) => {
+      return MetadataUtils.findElementByElementPath(store.editor.jsxMetadata, parentPath)
+    },
+    'NavigatorItem parent',
+  )
+
+  const isHiddenConditionalBranch = React.useMemo(() => {
+    const conditional = asConditional(parent)
+    if (conditional == null) {
+      return false
+    }
+    const flag = getConditionalFlag(conditional)
+    return flag == false
+      ? matchesOverriddenBranch(elementPath, parentPath, {
+          clause: conditional.whenTrue,
+          branch: 'then',
+          wantOverride: true,
+          parentOverride: true,
+        })
+      : matchesOverriddenBranch(elementPath, parentPath, {
+          clause: conditional.whenFalse,
+          branch: 'else',
+          wantOverride: false,
+          parentOverride: false,
+        })
+  }, [parent, elementPath, parentPath])
```

Comment:
could you turn this into a selector please? that way we don't re-render this row if the ElementInstanceMetadata of parent changes

Reply:
https://github.com/concrete-utopia/utopia/pull/3361/commits/7edbd68e2270e434c8704a6926684d7109887bf1

Reply:
@ruggi thank you! I'd only ask one more tweak: please make store the first parameter of the selectors – it's a convention that makes it easier to see store vs parameters when browsing the code for all our selectors

Reply:
🫡 https://github.com/concrete-utopia/utopia/pull/3361/commits/b8d619d765c326eb855139a3cf32193f1d78e8bb